opentelemetry-resourcedetector-gcp/src/opentelemetry/resourcedetector/gcp_resource_detector/_mapping.py

```diff
@@ -185,10 +194,24 @@ def _create_monitored_resource(
 
     for mr_key, map_config in mapping.items():
         mr_value = None
+        fallback_value = None
         for otel_key in map_config.otel_keys:
             if otel_key in resource_attrs:
-                mr_value = resource_attrs[otel_key]
-                break
+                if (
```

Comment:
Do you think it would be cleaner to move this out of the loop and do lazily directly at the L213 check? Either check by `if mr_key == _constants.TASK_ID and ...` or maybe `if mr_value.startswith(_constants.UNKNOWN_SERVICE_PREFIX)`.

Reply:
moved.  It does look cleaner.